### PR TITLE
scripts(lint-packages): add end of line check

### PR DESCRIPTION
**Pull request by truboxl**

### CODE REVIEW

**Comment by agnostic-apollo** [comment]
Actually all script/source files should end with a newline, not having one is against posix standard and will mess with editor compatibility.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Comment by truboxl** [comment]
Naming things short is hard...

**Comment by TomJo2000** [comment]
Apparently we don't enforce [`insert_final_newline`](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#insert_final_newline) in the `.editorconfig`
https://github.com/termux/termux-packages/blob/887e52732efe5c3d7470c38f798d8414f71fd135/.editorconfig#L1-L3
https://github.com/termux/termux-packages/blob/887e52732efe5c3d7470c38f798d8414f71fd135/.editorconfig#L7-L9

But as mentioned above, we'd probably want to for some filetypes.

File: root-packages/docker/build.sh [hunk 1]
```diff
@@ -118,9 +118,9 @@ termux_step_make_install() {
 	mkdir -p "${TERMUX_PREFIX}"/etc/docker
 	sed -e "s|@TERMUX_PREFIX@|$TERMUX_PREFIX|g" \
 		"${TERMUX_PKG_BUILDER_DIR}"/daemon.json > "${TERMUX_PREFIX}"/etc/docker/daemon.json
-        chmod 600 "${TERMUX_PREFIX}"/etc/docker/daemon.json
```
Comment (TomJo2000) [inline root-packages/docker/build.sh:121]: I guess I never checked the root packages for whitespace issues?

File: packages/alpine/build.sh [hunk 2]
```diff
@@ -56,4 +56,3 @@ termux_step_create_debscripts() {
 	echo "touch \$HOME/.pine-passfile" >> postinst
 	echo "fi" >> postinst
 }
-
```
Comment (TomJo2000) [inline packages/alpine/build.sh:59]: As mentioned my Apollo, a single newline at the end of the file is the POSIX standard, so these shouldn't be removed.

File: x11-packages/fcitx5-hangul/build.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@ TERMUX_PKG_LICENSE="LGPL-2.1"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="5.1.3"
 TERMUX_PKG_SRCURL="https://github.com/fcitx/fcitx5-hangul/archive/refs/tags/${TERMUX_PKG_VERSION}.tar.gz"
-TERMUX_PKG_SHA256=44fa27b0d0168f9e41d24da3e0f3818e09c15c11f2f99261d13134bcef478e45v
```
Comment (TomJo2000) [inline x11-packages/fcitx5-hangul/build.sh:7]: How did that even pass the linter with a `v` at the of the SHA?
https://github.com/termux/termux-packages/blob/887e52732efe5c3d7470c38f798d8414f71fd135/scripts/lint-packages.sh#L329-L336

**Review by TomJo2000 — CHANGES_REQUESTED** [review]
Changes to the linter look fine, we we'll only want to remove double newlines from package builds.

File: packages/alpine/build.sh (hunk 2 above)
Comment (truboxl) [reply packages/alpine/build.sh:59]: There is a newline like sylirre mentioned but it really depends on the editor to show it, for example gedit does show it.
The file libandroid-execinfo/LICENSE does not have a newline at the end and that's somebody need to fix it.
Technically we should check U+000A which is now incorporated into the linter.

**Comment by truboxl** [comment]
###### Emacs is superior because it shows the last newline while other editors don't /s

https://stackoverflow.com/questions/19240082/why-do-gedit-and-vim-hide-the-final-newline-from-the-user

I will merge by tomorrow. Its a common mistake to find the existence of final newline using editors so I don't blame anyone.

**Review by TomJo2000 — APPROVED** [review]

**Pull request merged**